src/index.ts

```diff
@@ -1,61 +1,171 @@
 import axios from 'axios';
 
-type SuggestResult = [
+type GoogleSuggestResult = [
   string,
   string[],
   [],
   string[],
   {
-    'google:clientdata': { 'bpc':boolean, 'tlw':boolean },
+    'google:clientdata': { 'bpc': boolean, 'tlw': boolean },
     'google:suggestrelevance': number[],
     'google:suggestsubtypes': number[][],
     'google:suggesttype': string[],
     'google:verbatimrelevance': number
   },
 ];
 
+type DuckDuckGoSuggestResult = {
+  'phrase': string
+}[]
+
+type YahooSuggestResult = {
+  q: string,
+  l: {
+    gprid: string
+  },
+  r: {
+    'k': string,
+    'm': number
+  }[]
+}
 type SuggestionType = 'QUERY' | 'NAVIGATION';
+type SearchEngine = 'Google' | 'Yahoo' | 'DuckDuckGo' | 'random'
+/**
+ * Gets search suggestions for a partial search
+ *
+ * @export
+ * @param {string} partialSearch The term to search suggestions for
+ * @param {SearchEngine} searchEngine
+ * @return {Promise<Suggestion[]>} The suggested searches
+ */
+ export async function getSuggestions(partialSearch: string, searchEngine: SearchEngine = "Google"): Promise<Suggestion[]> {
+  if (searchEngine == 'random') {
+    const engines = ['Google','Yahoo','DuckDuckGo']
+    searchEngine = engines[Math.floor(Math.random()*engines.length)] as SearchEngine
+  }
+  switch (searchEngine) {
+    case 'DuckDuckGo':
+      return new DuckDuckGo().getSuggestions(partialSearch)
+    case 'Yahoo':
+      return new Yahoo().getSuggestions(partialSearch)
+    default:
+      return new Google().getSuggestions(partialSearch)
+  }
+};
 
 export interface Suggestion {
   term: string,
   type: SuggestionType
 }
 
-/**
- * Gets the URL to query the autosuggest service
- *
- * @param {string} searchTerm
- * @return {string}
- */
-function getUrl(searchTerm: string): string {
-  return `http://suggestqueries.google.com/complete/search?client=chrome&q=${searchTerm}`;
+interface BaseSearch {
```

Comment:
If we end up moving providers to its own folders / files, maybe this could be `src/providers/base.ts` and the rest implement this BaseProvider interface?